File: cpp/Pair.cpp
#include "Pair.hh"

Pair::Pair()
{
    matrix = Matrix(0);
    detetminant = 0.0;
}

Pair::Pair(Matrix _matrix)
{
    matrix = _matrix;
    detetminant = 0.0;
}

Pair::Pair(Matrix _matrix, double det)
{
    matrix = _matrix;
    detetminant = det;
}

std::string Pair::ToString()
{
    std::string result  = matrix.ToString();
    result += "Det: ";
    result += std::to_string(detetminant);
    result += "\n";

    return result;
}

File: dotnet/Pair.cs
namespace dotnet
{
    public class Pair
    {
        public Matrix Matrix { get; set; }
        public double Determinant { get; set; }

        public Pair(Matrix matrix)
        {
            Matrix = matrix;
            Determinant = 0.0;
        }

        public override string ToString()
        {
            string result = this.Matrix.ToString();
            result += "Det: ";
            result += Determinant.ToString();
            result += "\n";

            return result;
        } 
    }
}